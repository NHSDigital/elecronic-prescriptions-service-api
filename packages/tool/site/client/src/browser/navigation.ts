export function redirect(url: string): void {
  // eslint-disable-next-line no-undef
  window.location.assign(url)
}

export interface Redirect {
  redirectUri: string
}

export function isRedirect(data: unknown): data is Redirect {
  return (data as Redirect).redirectUri !== undefined
}
