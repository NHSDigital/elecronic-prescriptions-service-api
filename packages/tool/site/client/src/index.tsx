import * as React from "react"
import {Page} from "./components/page"
import SignPage from "./pages/signPage"
import * as ReactDOM from "react-dom"
import {BrowserRouter, Route, Switch} from "react-router-dom"
import ClaimPage from "./pages/claimPage"
import DispensePage from "./pages/dispensePage"
import {CookiesProvider} from "react-cookie"
import SendPage from "./pages/sendPage"
import HomePage from "./pages/homePage"
import PrescriptionSearchPage from "./pages/prescriptionSearchPage"
import ReleasePage from "./pages/releasePage"
import CancelPage from "./pages/cancelPage"
import MyPrescriptionsPage from "./pages/myPrescriptionsPage"
import LoginPage from "./pages/loginPage"
import ValidatePage from "./pages/validatePage"
import ReturnPage from "./pages/returnPage"
import WithdrawPage from "./pages/withdrawPage"
import LoadPage from "./pages/loadPage"
import {Environment} from "./services/environment"
import LogoutPage from "./pages/logoutPage"
import ConfigPage from "./pages/configPage"
import ComparePage from "./pages/comparePage"
import ViewPrescriptionPage from "./pages/viewPrescriptionPage"
import {PageFullWidth} from "./components/pageFullWidth"
import {PageContainer} from "./components/pageContainer"
import DoseToTextPage from "./pages/doseToTextPage"
import TrackerViewPrescriptionPage from "./pages/trackerViewPrescription"

const customWindow = window as Record<string, any>

export interface AppContextValue {
  baseUrl: string
  environment: Environment
}

export const AppContext = React.createContext<AppContextValue>({baseUrl: "/", environment: "int"})

async function startApplication(baseUrl: string, environment: Environment): Promise<void> {
  const urlParams = new URLSearchParams(window.location.search)
  const content = (
    <AppContext.Provider value={{baseUrl, environment}}>
      <CookiesProvider>
        <Page>
          <BrowserRouter>
            <Switch>
              <Route path={`${baseUrl}/`}>
                <PageContainer>
                  <HomePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dose-to-text`}>
                <PageContainer>
                  <DoseToTextPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}config`}>
                <PageContainer>
                  <ConfigPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}my-prescriptions`}>
                <PageContainer>
                  <MyPrescriptionsPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}login`}>
                <PageContainer>
                  <LoginPage separateAuth={urlParams.get("separate_auth")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}logout`}>
                <PageContainer>
                  <LogoutPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/load`}>
                <PageContainer>
                  <LoadPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/edit`}>
                <PageContainer>
                  <SignPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/send`}>
                <PageContainer>
                  <SendPage token={urlParams.get("token")} state={urlParams.get("state")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/cancel`}>
                <PageContainer>
                  <CancelPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/release`}>
                <PageContainer>
                  <ReleasePage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/return`}>
                <PageContainer>
                  <ReturnPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/withdraw`}>
                <PageContainer>
                  <WithdrawPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/dispense`}>
                <PageContainer>
                  <DispensePage prescriptionId={urlParams.get("prescription_id")} amendId={urlParams.get("amend_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/claim`}>
                <PageContainer>
                  <ClaimPage prescriptionId={urlParams.get("prescription_id")} amend={!!urlParams.get("amend")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}search`}>
                <PageContainer>
                  <PrescriptionSearchPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}tracker`}>
                <PageContainer>
                  <TrackerViewPrescriptionPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}view`}>
                <PageContainer>
                  <ViewPrescriptionPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}validate`}>
                <PageContainer>
                  <ValidatePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}compare-prescriptions`}>
                <PageFullWidth>
                  <ComparePage />
                </PageFullWidth>
              </Route>
            </Switch>
          </BrowserRouter>
        </Page>
      </CookiesProvider>
    </AppContext.Provider>
  )
  ReactDOM.render(content, document.getElementById("root"))
}

customWindow.startApplication = startApplication
