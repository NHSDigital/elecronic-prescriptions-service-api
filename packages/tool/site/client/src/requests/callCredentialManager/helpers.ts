// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
const element = (identifier: string) => document.getElementById(identifier)!

export interface HubResponse {
    certificate: string,
    status_code: number,
    status_string: string,
    message: string,
    timestamp: string,
    signatures: Array<Signature>,
    failed_signatures: Array<Signature>
  }

export interface Signature {
    id: string,
    signature: string
  }

export async function start(jwt: string, signFn: (jwt: string) => Promise<HubResponse>): Promise<void> {
  try {
    //showStatus("Signing in progress", "Communicating with Credential Management to generate a signature.")
    const hubResponse = await signFn(jwt)
    validateHubResponse(hubResponse)

  } catch (e) {
    console.log(e)
    showError(e.message)
  }
}

function validateHubResponse(hubResponse: HubResponse) {
  if (!hubResponse) {
    throw new TypeError("No response from Credential Management.")
  }

  if (hubResponse.status_code !== 0) {
    console.log(`Got error response from Credential Management: ${JSON.stringify(hubResponse)}`)
    throw new TypeError("Error response from Credential Management.")
  }

  if (!hubResponse.certificate || !hubResponse.signatures?.length) {
    console.log(`Got invalid response from Credential Management: ${JSON.stringify(hubResponse)}`)
    throw new TypeError("Invalid response from Credential Management.")
  }
}

export function showError(details?: string): void {
  element("status-container").style.display = "none"
  if (details) {
    element("error-details").innerText = `Error details: ${details}`
  }
  element("error-container").style.display = "block"
}
