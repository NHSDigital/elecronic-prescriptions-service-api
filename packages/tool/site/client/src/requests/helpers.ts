export interface HubResponse {
  certificate: string,
  status_code: number,
  status_string: string,
  message: string,
  timestamp: string,
  signatures: Array<Signature>,
  failed_signatures: Array<Signature>
}

export interface Signature {
  id: string,
  signature: string
}

export function getToken(queryString: URLSearchParams): string {
  const token = queryString.get("token")

  if (!token) {
    throw new TypeError("Token not present in query string.")
  }

  return token
}

export function getState(queryString: URLSearchParams): string | null {
  return queryString.get("state")
}

export function getQueryString(): URLSearchParams {
  const queryString = window.location.search
  if (!queryString) {
    throw new TypeError("Empty query string.")
  }

  return new URLSearchParams(queryString)
}

export async function start(signFn: (jwt: string) => Promise<HubResponse>): Promise<void> {
  try {
    //showStatus("Signing in progress", "Downloading the signature request from the Signing Service.")
    const queryString = getQueryString()
    const token = getToken(queryString)
    const signatureRequest = await retrieveSignatureRequestFromServer(token)

    //showStatus("Signing in progress", "Communicating with Credential Management to generate a signature.")
    const hubResponse = await signFn(signatureRequest)
    validateHubResponse(hubResponse)

    //showStatus("Signing in progress", "Uploading the signature to the Signing Service.")
    const redirectUrl = await sendResponseToProvider(hubResponse, token)

    //showStatus("Signing complete", "Sending you back to the calling application.")
    const state = getState(queryString)
    window.location.href = state ? `${redirectUrl}&state=${state}` : redirectUrl
  } catch (e) {
    console.log(e)
    showError(e.message)
  }
}

function validateHubResponse(hubResponse: HubResponse) {
  if (!hubResponse) {
    throw new TypeError("No response from Credential Management.")
  }

  if (hubResponse.status_code !== 0) {
    console.log(`Got error response from Credential Management: ${JSON.stringify(hubResponse)}`)
    throw new TypeError("Error response from Credential Management.")
  }

  if (!hubResponse.certificate || !hubResponse.signatures?.length) {
    console.log(`Got invalid response from Credential Management: ${JSON.stringify(hubResponse)}`)
    throw new TypeError("Invalid response from Credential Management.")
  }
}

async function retrieveSignatureRequestFromServer(token: string): Promise<string> {
  const response = await fetch(`./client/signaturerequest/${token}`, {
    headers: {
      "X-CSRF-Token": getCSRFTokenFromCookie()
    }
  })
  if (response.status !== 200) {
    console.error(`Got unexpected response from Signing Service: ${response.status}`)
    throw new Error("Failed to retrieve signature request from Signing Service.")
  }
  const responseText = await response.text()
  if (!responseText) {
    throw new Error("Failed to parse response from Signing Service.")
  }
  return responseText
}

function getCSRFTokenFromCookie(): string {
  const cookies = getCookies()
  const crumb = cookies.get("crumb")

  if (!crumb) {
    throw new Error("Could not find CSRF token")
  }

  return crumb
}

function getCookies(): Map<string, string> {
  try {
    return new Map(document.cookie.split(";").map(cookie => {
      const cookieKey = cookie.split("=")[0].trim()
      const cookieValue = cookie.split("=")[1].trim()

      return [cookieKey, cookieValue]
    }))
  } catch(e) {
    console.error("Could not read cookies")
    console.error(e)
    return new Map()
  }
}

// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
const element = (identifier: string) => document.getElementById(identifier)!

export function showError(details?: string): void {
  element("status-container").style.display = "none"
  if (details) {
    element("error-details").innerText = `Error details: ${details}`
  }
  element("error-container").style.display = "block"
}

export async function sendResponseToProvider(hubResponse: HubResponse, token: string): Promise<string> {
  const response = await fetch(
    `./client/signatureresponse/${token}`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "X-CSRF-Token": getCSRFTokenFromCookie()
      },
      body: JSON.stringify({
        "signatures": hubResponse.signatures,
        "certificate": hubResponse.certificate
      })
    }
  )

  if (response.status !== 200) {
    console.error(`Got unexpected response from Signing Service: ${response.status}`)
    throw new Error("Failed to upload signature to Signing Service.")
  }

  const responseJson = await response.json()
  if (!responseJson) {
    throw new Error("Failed to parse response from Signing Service.")
  }

  return responseJson.redirectUri
}
