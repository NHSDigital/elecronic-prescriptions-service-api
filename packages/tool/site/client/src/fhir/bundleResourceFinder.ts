import * as fhir from "fhir/r4"
import {MedicationDispense, MedicationRequest} from "./helpers"

export const getPatientResources = buildResourceFinder<fhir.Patient>("Patient")
export const getPractitionerResources = buildResourceFinder<fhir.Practitioner>("Practitioner")
export const getPractitionerRoleResources = buildResourceFinder<fhir.PractitionerRole>("PractitionerRole")
export const getOrganizationResources = buildResourceFinder<fhir.Organization>("Organization")
export const getMedicationRequestResources = buildResourceFinder<MedicationRequest>("MedicationRequest")
export const getMedicationDispenseResources = buildResourceFinder<MedicationDispense>("MedicationDispense")
export const getMessageHeaderResources = buildResourceFinder<fhir.MessageHeader>("MessageHeader")
export const getCommunicationRequestResources = buildResourceFinder<fhir.CommunicationRequest>("CommunicationRequest")

function buildResourceFinder<T extends fhir.FhirResource>(resourceType: string): (payload: fhir.Bundle) => Array<T> {
  return function (bundle: fhir.Bundle): Array<T> {
    const typeCheck = (resource: fhir.FhirResource): resource is T => resource.resourceType === resourceType
    return bundle.entry.map(entry => entry.resource).filter(typeCheck)
  }
}
