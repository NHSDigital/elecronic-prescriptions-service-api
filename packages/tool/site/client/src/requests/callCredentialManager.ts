import {HubResponse, start} from "./helpers"

// eslint-disable-next-line @typescript-eslint/no-explicit-any
(window as any).jQuery = jQuery;
// eslint-disable-next-line @typescript-eslint/no-explicit-any
(window as any).$ = jQuery

const SIGNALR_URL = `"http://localhost:"+prService.portNumber()+"/signalr"`
const SIGNALR_HUB_NAME = "signingHub"
const SIGNALR_METHOD_NAME = "requestToSign"

export async function sign(jwt: string): Promise<HubResponse> {
  const connection = $.hubConnection(SIGNALR_URL, {logging: true})
  if (connection.state !== SignalR.ConnectionState.Disconnected) {
    console.log(`Unexpected SignalR connection state ${connection.state}`)
    throw new Error("SignalR connection is already active.")
  }

  try {
    const proxy = connection.createHubProxy(SIGNALR_HUB_NAME)
    await connection.start({transport: "longPolling"})
    return await proxy.invoke(SIGNALR_METHOD_NAME, jwt)
  } finally {
    connection.stop()
  }
}

window.addEventListener("DOMContentLoaded", () => {
  start(sign)
})
