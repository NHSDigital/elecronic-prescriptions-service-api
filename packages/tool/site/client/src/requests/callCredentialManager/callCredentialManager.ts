import {HubResponse} from "./helpers"

import $ from "jquery"

if (!(window as any).jQuery) {
  (window as any).jQuery = $;
  (window as any).$ = $
}

import "signalr"

const SIGNALR_HUB_NAME = "signingHub"
const SIGNALR_METHOD_NAME = "requestToSign"

export async function sign(jwt: string): Promise<HubResponse> {
  const SIGNALR_URL = "http://localhost:"+(window as any).prService.portNumber()+"/signalr"
  console.log("SignalR Url: " + SIGNALR_URL)
  console.log("JWT: " + jwt)
  const connection = (window as any).$.hubConnection(SIGNALR_URL, {logging: true})
  if (connection.state !== (window as any).$.signalR.connectionState.disconnected) {
    console.log(`Unexpected SignalR connection state ${connection.state}`)
    throw new Error("SignalR connection is already active.")
  }

  try {
    const proxy = connection.createHubProxy(SIGNALR_HUB_NAME)
    await connection.start({transport: "longPolling"})
    return await proxy.invoke(SIGNALR_METHOD_NAME, jwt)
  } finally {
    connection.stop()
  }
}
