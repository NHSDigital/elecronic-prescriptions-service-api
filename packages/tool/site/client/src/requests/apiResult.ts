import {FhirResource} from "fhir/r4"

export function isApiResult(data: unknown): data is ApiResult {
  const result = data as ApiResult
  return typeof result.success === "boolean"
    && "request" in result
    && "response" in result
}

export interface ApiResult {
  success: boolean
  request: FhirResource
  request_xml?: string
  response: FhirResource
  response_xml?: string
}
