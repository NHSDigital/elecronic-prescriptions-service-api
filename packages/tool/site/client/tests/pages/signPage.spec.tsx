import {waitFor} from "@testing-library/react"
import {screen} from "@testing-library/dom"
import pretty from "pretty"
import * as React from "react"
import moxios from "moxios"
import userEvent from "@testing-library/user-event"
import {readBundleFromFile} from "../messages"
import {AppContextValue} from "../../src"
import {renderWithContext} from "../renderWithContext"
import SignPage from "../../src/pages/signPage"
import {OperationOutcome} from "fhir/r4"
import {redirect} from "../../src/browser/navigation"
import {axiosInstance} from "../../src/requests/axiosInstance"
import {MomentInput} from "moment"
import {internalDev} from "../../src/services/environment"
import {sign} from "../../src/requests/callCredentialManager"

const baseUrl = "baseUrl/"
const context: AppContextValue = {baseUrl, environment: internalDev}

const prescriptionsUrl = `${baseUrl}prescriptions`
const editPrescriptionsUrl = `${baseUrl}prescribe/edit`
const signatureRequestUrl = `${baseUrl}sign/upload-signatures`

const prescriptionOrder = readBundleFromFile("prescriptionOrder.json")

jest.mock("moment", () => {
  const actualMoment = jest.requireActual("moment")
  return ({
    ...actualMoment,
    utc: (inp?: MomentInput, strict?: boolean) => actualMoment.utc(inp ?? "2021-11-13T10:57:13.000Z", strict)
  })
})

jest.mock("../../src/browser/navigation")
jest.mock("../../src/requests/callCredentialManager")
jest.mock("../../src/requests/helpers")

beforeEach(() => moxios.install(axiosInstance))

afterEach(() => moxios.uninstall(axiosInstance))

test("Displays loading text while prescription data is being requested", async () => {
  const {container} = renderWithContext(<SignPage/>, context)
  await waitFor(() => screen.getByText("Retrieving prescription details."))

  expect(screen.getByText("Retrieving prescription details.")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays prescription summary if prescription details are retrieved successfully", async () => {
  moxios.stubRequest(prescriptionsUrl, {
    status: 200,
    response: [prescriptionOrder]
  })

  const container = await renderPage()

  expect(screen.getByText("Sign & Send")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays loading text while prescription is being sent", async () => {
  moxios.stubRequest(prescriptionsUrl, {
    status: 200,
    response: [prescriptionOrder]
  })

  const container = await renderPage()
  userEvent.click(screen.getByText("Sign & Send"))
  await waitFor(() => screen.getByText("Sending signature request."))

  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Calls to Credential Management", async () => {
  moxios.stubRequest(signatureRequestUrl, {
    status: 200,
    response: {
      redirectUri: "https://example.com/"
    }
  })
  moxios.stubRequest(prescriptionsUrl, {
    status: 200,
    response: [prescriptionOrder]
  })
  moxios.stubRequest(editPrescriptionsUrl, {
    status: 200,
    response: {
      redirectUri: ""
    }
  })
  await renderPage()
  userEvent.click(screen.getByText("Sign & Send"))

  await waitFor(() => screen.getByText("Upload Complete"))

  expect(sign).toHaveBeenCalled()
})

test("Redirects and displays link if signature request upload is successful", async () => {
  moxios.stubRequest(signatureRequestUrl, {
    status: 200,
    response: {
      redirectUri: "https://example.com/"
    }
  })
  moxios.stubRequest(prescriptionsUrl, {
    status: 200,
    response: [prescriptionOrder]
  })
  moxios.stubRequest(editPrescriptionsUrl, {
    status: 200,
    response: {
      redirectUri: ""
    }
  })
  const container = await renderPage()
  userEvent.click(screen.getByText("Sign & Send"))
  await waitFor(() => screen.getByText("Upload Complete"))

  expect(redirect).toHaveBeenCalledWith("https://example.com/")

  const link = screen.getByRole<HTMLAnchorElement>("link")
  expect(link.text).toEqual("Proceed to the Signing Service")
  expect(link.href).toEqual("https://example.com/")
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays error message if prepare errors present", async () => {
  const operationOutcome: OperationOutcome = {
    resourceType: "OperationOutcome",
    issue: [{
      severity: "fatal",
      code: "invalid",
      diagnostics: "Some error message"
    }]
  }
  moxios.stubRequest(signatureRequestUrl, {
    status: 200,
    response: {
      prepareErrors: [operationOutcome]
    }
  })
  moxios.stubRequest(prescriptionsUrl, {
    status: 200,
    response: [prescriptionOrder]
  })
  moxios.stubRequest(editPrescriptionsUrl, {
    status: 200,
    response: {
      redirectUri: ""
    }
  })

  //const container = await renderPage()
  userEvent.click(screen.getByText("Sign & Send"))
  // await waitFor(() => screen.getByText("Error"))

  // expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays error message if redirect URI not present", async () => {
  const operationOutcome: OperationOutcome = {
    resourceType: "OperationOutcome",
    meta: {
      lastUpdated: "2022-10-21T13:47:00+00:00"
    },
    issue: [{
      severity: "fatal",
      code: "invalid",
      diagnostics: "Some error message"
    }]
  }
  moxios.stubRequest(signatureRequestUrl, {
    status: 400,
    statusText: "Bad Request",
    response: operationOutcome
  })
  moxios.stubRequest(prescriptionsUrl, {
    status: 200,
    response: [prescriptionOrder]
  })
  moxios.stubRequest(editPrescriptionsUrl, {
    status: 200,
    response: {
      redirectUri: ""
    }
  })

  //const container = await renderPage()
  userEvent.click(screen.getByText("Sign & Send"))
  // await waitFor(() => screen.getByText("Error"))

  // expect(pretty(container.innerHTML)).toMatchSnapshot()
})

async function renderPage() {
  const {container} = renderWithContext(<SignPage/>, context)
  await waitFor(() => screen.getByText("Prescription Summary"))
  return container
}
