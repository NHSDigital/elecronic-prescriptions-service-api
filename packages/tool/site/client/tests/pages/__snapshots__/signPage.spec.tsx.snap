// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays loading text while prescription data is being requested 1`] = `
"<div style="text-align: center;">
  <h1 class="nhsuk-label-wrapper"><label class="nhsuk-label nhsuk-label--xl sc-gLDzan fnXRyb">Loading</label></h1>
  <div class="nhsuk-loader"><span class="nhsuk-loader__spinner"></span><span class="nhsuk-loader__text">Retrieving prescription details.</span></div>
</div>"
`;

exports[`Displays loading text while prescription is being sent 1`] = `
"<div style="text-align: center;">
  <h1 class="nhsuk-label-wrapper"><label class="nhsuk-label nhsuk-label--xl sc-gLDzan fnXRyb">Loading</label></h1>
  <div class="nhsuk-loader"><span class="nhsuk-loader__spinner"></span><span class="nhsuk-loader__text">Sending signature request.</span></div>
</div>"
`;

exports[`Displays prescription summary if prescription details are retrieved successfully 1`] = `
"<form>
  <h1 class="nhsuk-label-wrapper"><label class="nhsuk-label nhsuk-label--xl"><span>Prescription Summary</span>
      <figure class="nhsuk-image"><img class="nhsuk-image__img sc-hLseeU jvNVTN" id="editPrescription" srcset="baseUrl/static/BlackTie_Bold_full_set_Pencil_SVG_Blue.svg" sizes="50px"></figure>
    </label></h1>
  <div class="sc-dmqHEX exmKrI"><button class="nhsuk-button" aria-disabled="false" type="button">Download this Prescription</button></div>
  <dl class="nhsuk-summary-list">
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">ID</dt>
      <dd class="nhsuk-summary-list__value">B407F7-A99968-4DB2AU</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Prescription Type Code</dt>
      <dd class="nhsuk-summary-list__value">1001</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Course Of Therapy</dt>
      <dd class="nhsuk-summary-list__value">Continuous long term (repeat dispensing)</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Issue Number</dt>
      <dd class="nhsuk-summary-list__value">1 of 6</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Authored On</dt>
      <dd class="nhsuk-summary-list__value">13-Nov-2021</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Effective Date</dt>
      <dd class="nhsuk-summary-list__value">03-Sep-2021</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Nominated Pharmacy ODS Code</dt>
      <dd class="nhsuk-summary-list__value">FER21</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Nominated Pharmacy Type</dt>
      <dd class="nhsuk-summary-list__value">Other (e.g. Community Pharmacy)</dd>
    </div>
  </dl><label class="nhsuk-label nhsuk-label--m">Patient</label>
  <dl class="nhsuk-summary-list">
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Name</dt>
      <dd class="nhsuk-summary-list__value">CORY, ETTA (MISS)</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">NHS Number</dt>
      <dd class="nhsuk-summary-list__value">944 930 5552</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Date of Birth</dt>
      <dd class="nhsuk-summary-list__value">04-Jan-1999</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Gender</dt>
      <dd class="nhsuk-summary-list__value">Female</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Address</dt>
      <dd class="nhsuk-summary-list__value">123 Dale Avenue<br>Long Eaton<br>Nottingham<br>NG10 1NP</dd>
    </div>
  </dl>
  <div class="nhsuk-table__panel-with-heading-tab">
    <h3 class="nhsuk-table__heading-tab">Medication</h3>
    <table class="nhsuk-table">
      <caption class="nhsuk-table__caption">Medication summary</caption>
      <thead class="nhsuk-table__head">
        <tr class="nhsuk-table__row">
          <th class="nhsuk-table__header" scope="col">Details</th>
          <th class="nhsuk-table__header" scope="col">Quantity</th>
          <th class="nhsuk-table__header" scope="col">Unit</th>
          <th class="nhsuk-table__header" scope="col">Dosage Instructions</th>
        </tr>
      </thead>
      <tbody class="nhsuk-table__body">
        <tr class="nhsuk-table__row">
          <td class="nhsuk-table__cell">
            <div><b>Methotrexate 10mg/0.2ml solution for injection pre-filled syringes</b></div>
          </td>
          <td class="nhsuk-table__cell">63</td>
          <td class="nhsuk-table__cell">tablet</td>
          <td class="nhsuk-table__cell">
            <div>10 milligram, Inject, Subcutaneous route, once weekly</div>
          </td>
        </tr>
      </tbody>
    </table>
  </div><label class="nhsuk-label nhsuk-label--m">Prescriber</label>
  <dl class="nhsuk-summary-list">
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Name</dt>
      <dd class="nhsuk-summary-list__value">EDWARDS, Thomas (DR)</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Professional Codes</dt>
      <dd class="nhsuk-summary-list__value">SDS Role Profile ID - 100102238986<br>Spurious Code - 612345<br>SDS User ID - 3415870201<br>GMC Number - 4428981</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Telecom</dt>
      <dd class="nhsuk-summary-list__value">01234567890</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Organization</dt>
      <dd class="nhsuk-summary-list__value">SOMERSET BOWEL CANCER SCREENING CENTRE (A99968)</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Address</dt>
      <dd class="nhsuk-summary-list__value">MUSGROVE PARK HOSPITAL<br>TAUNTON<br>TA1 5DA</dd>
    </div>
    <div class="nhsuk-summary-list__row">
      <dt class="nhsuk-summary-list__key">Trust / CCG</dt>
      <dd class="nhsuk-summary-list__value">TAUNTON AND SOMERSET NHS FOUNDATION TRUST (RBA)</dd>
    </div>
  </dl>
  <div class="sc-dmqHEX exmKrI"><button class="nhsuk-button" aria-disabled="false" type="submit">Sign &amp; Send</button><button class="nhsuk-button nhsuk-button--secondary" aria-disabled="false" type="button">Back</button></div>
</form>"
`;
